torch/csrc/autograd/function_port.cpp

```diff
@@ -0,0 +1,38 @@
+#include "torch/csrc/autograd/function_port.h"
+
+#include "torch/csrc/autograd/function.h"
+#include "torch/csrc/autograd/variable.h"
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+
+namespace torch {
+namespace autograd {
+FunctionPort FunctionPort::for_gradient(const Variable& variable) {
```

Comment:
I think this makes more sense as a method on variable, right? `v.gradient_edge()`

Reply:
Isn't there already a method like this?

Reply:
I would recommend only looking at the most recent commit in this PR. These three commits are three separate PRs that gradually implement the change @zdevito  suggests.

Reply:
Oh, sorry I missed that this was three commits. It is hard to read just the changes this PR introduces since github doesn't let you view a range of commits (as far as I know)